classes/sushi/IR.php

```diff
@@ -371,4 +378,254 @@ public function getReportItems(): array
 
         return $items;
     }
+
+    /** Get TSV report column names */
+    public function getTSVColumnNames(): array
+    {
+        $columnRow = ['Item', 'Publisher', 'Publisher ID', 'Platform'];
+
+        if (in_array('Authors', $this->attributesToShow)) {
+            array_push($columnRow, 'Authors');
+        }
+        if (in_array('Publication_Date', $this->attributesToShow)) {
+            array_push($columnRow, 'Publication_Date');
+        }
+        if (in_array('Article_Version', $this->attributesToShow)) {
+            array_push($columnRow, 'Article_Version');
+        }
+
+        array_push($columnRow, 'DOI', 'Proprietary_ID', 'ISBN', 'Print_ISSN', 'Online_ISSN', 'URI');
+
+        if ($this->includeParentDetails == 'True') {
+            array_push(
+                $columnRow,
+                'Parent_Title',
+                'Parent_Authors',
+                'Parent_Publication_Date',
+                'Parent_Article_Version',
+                'Parent_Data_Type',
+                'Parent_DOI',
+                'Parent_Proprietary_ID',
+                'Parent_ISBN',
+                'Parent_Print_ISSN',
+                'Parent_Online_ISSN',
+                'Parent_URI'
+            );
+        }
+
+        if (in_array('Data_Type', $this->attributesToShow)) {
+            array_push($columnRow, 'Data_Type');
+        }
+        if (in_array('YOP', $this->attributesToShow)) {
+            array_push($columnRow, 'YOP');
+        }
+        if (in_array('Access_Type', $this->attributesToShow)) {
+            array_push($columnRow, 'Access_Type');
+        }
+        if (in_array('Access_Method', $this->attributesToShow)) {
+            array_push($columnRow, 'Access_Method');
+        }
+
+        array_push($columnRow, 'Metric_Type', 'Reporting_Period_Total');
+
+        if ($this->granularity == 'Month') {
+            $period = $this->getMonthlyDatePeriod();
+            foreach ($period as $dt) {
+                array_push($columnRow, $dt->format('M-Y'));
+            }
+        }
+
+        return [$columnRow];
+    }
+
+    /** Get TSV report rows */
+    public function getTSVReportItems(): array
+    {
+        $results = $this->getQueryResults();
+
+        $resultsGroupedBySubmission = $resultRows = [];
```

Comment:
`$resultsGroupedBySubmission` is immediately overwritten, so this line can be removed.

Reply:
Same as above about only removing `$resultsGroupedBySubmission`, not `$resultRows`.